File: vjudge/2020七月暑假集训/Nowcoder 多校 2020/Day 7/D.java
import java.math.BigInteger;
import java.util.*;

public class Main
{
    final static int N = (int)(1e5 + 5);
    final static BigInteger zero = new BigInteger("0");
    final static String yes = "Fake news!";
    final static String no = "Nobody knows it better than me!";
    final static BigInteger one = BigInteger.valueOf(1);
    final static BigInteger two = BigInteger.valueOf(2);
    final static BigInteger six = BigInteger.valueOf(6);

    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t --> 0)
        {
            BigInteger n = sc.nextBigInteger();
            BigInteger sum = n.multiply(n.add(one)).multiply(n.multiply(two).add(one)).divide(six);
            BigInteger sq = sum.sqrt();
            BigInteger sq2 = sq.multiply(sq);
            if (sq2.equals(sum)) System.out.println(yes);
            else System.out.println(no);
        }
    }
}

File: vjudge/2020七月暑假集训/Nowcoder 多校 2020/Day 7/D.py
import math

t = int(input())
for i in range(t):
    n = int(input())
    sum = n * (n + 1) * (2 * n + 1) // 6
    sq = int(math.sqrt(sum))
    if sq * sq == sum :
        print("Fake news!")
    else:
        print("Nobody knows it better than me!")
